smart/map_processing.py

```diff
@@ -0,0 +1,171 @@
+import numpy as np
+import skimage as ski
+from matplotlib import colors
+from skimage.morphology import disk
+
+import astropy.units as u
+
+from sunpy.map import Map, all_coordinates_from_map, coordinate_is_on_solar_disk
+
+
+def map_processing(file):
+    """
+    Creating a map from a fits file and processing said map.
+
+    Parameters
+    ----------
+    file : fits
+        fits file to turn into map.
+
+    Returns
+    -------
+    im_map : sunpy.map.Map
+        processed map created from fits file.
+
+    """
+    im_map = Map(file)
+    if im_map.meta["CROTA2"] != 0:
+        im_map = im_map.rotate(order=3)
+    im_map.data[~coordinate_is_on_solar_disk(all_coordinates_from_map(im_map))] = np.nan
+    im_map.cmap.set_bad("k")
+    im_map.plot_settings["norm"] = colors.Normalize(vmin=-200, vmax=200)
+    return im_map
+
+
+@u.quantity_input
+def STL(
+    im_map,
+    thresh=100,
+    dilation_radius: u.arcsec = 9000 * u.arcsec,
```

Comment:
```suggestion
    dilation_radius: Quantity[u.arcsec] = 25 * u.deg,
```
and similar for the rest